application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadAutoArchiver.java

```diff
@@ -122,18 +126,35 @@ private static boolean shouldBeArchived(MessageChannel channel, Instant archiveA
     }
 
     private void handleArchiveFlow(ThreadChannel threadChannel, MessageEmbed embed) {
-        Consumer<Throwable> handleFailure = error -> {
-            if (error instanceof ErrorResponseException) {
-                logger.warn("Unknown error occurred during help thread auto archive routine",
-                        error);
-            }
-        };
+
+        Function<Result<Member>, RestAction<Message>> sendEmbedWithMention =
+                member -> threadChannel.sendMessage(member.get().getAsMention()).addEmbeds(embed);
+
+        Supplier<RestAction<Message>> sendEmbedWithoutMention =
+                () -> threadChannel.sendMessageEmbeds(embed);
 
         threadChannel.getGuild()
             .retrieveMemberById(threadChannel.getOwnerIdLong())
-            .flatMap(author -> threadChannel.sendMessage(author.getAsMention()).addEmbeds(embed))
-            .flatMap(any -> threadChannel.getManager().setArchived(true))
-            .queue(any -> {
-            }, handleFailure);
+            .mapToResult()
+            .flatMap(foundMember -> {
+                if (foundMember.isSuccess()) {
+                    return sendEmbedWithMention.apply(foundMember);
+                }
+                logger.info(
+                        "Owner of thread with id: {} left the server, sending embed without mention",
+                        threadChannel.getId(), foundMember.getFailure());
+
+                return sendEmbedWithoutMention.get();
+            })
+            .mapToResult()
+            .flatMap(sentEmbed -> {
+                if (sentEmbed.isFailure()) {
+                    logger.warn(
+                            "Unknown error occurred during help thread auto archive routine, archiving thread",
+                            sentEmbed.getFailure());
+                }
```

Comment:
well, in what case can embed sending fail? now that u handle user not fetchable separately. Ig we can safely remove this part

Reply:
it's supposed to be catch it all, just to be at least aware if something fails on sending message that can be handled in future. Are you sure?

Reply:
well, we never check if a message sending failed so... remove it

Reply:
fair